apps/music/test/marionette/loading_test.js

```diff
@@ -0,0 +1,78 @@
+/* global require, marionette, setup, suite, test */
+'use strict';
+
+var assert = require('assert');
+var Music = require('./lib/music.js');
+
+marionette('Music files loading', function() {
+  var apps = {};
+
+  var client = marionette.client({
+    prefs: {
+      'device.storage.enabled': true,
+      'device.storage.testing': true,
+      'device.storage.prompt.testing': true
+    },
+
+    settings: {
+      'lockscreen.enabled': false,
+      'ftu.manifestURL': null
+    },
+
+    apps: apps
+  });
+
+  var music;
+
+  setup(function() {
+    music = new Music(client);
+
+  });
+
+  suite('Detection of movies.', function () {
+    setup(function() {
+      client.fileManager.removeAllFiles();
+      client.fileManager.add([
+        {
+          // Regular Ogg Vorbis.
+          type: 'music',
+          filePath: 'apps/music/test-data/playlists/a.ogg'
+        },
+        {
+          // AAC file.
+          type: 'music',
+          filePath: 'apps/music/test-data/aac-tags.mp4'
+        },
+        {
+          // MPEG4 video file. Shouldn't load.
+          type: 'music',
+          filePath: 'test_media/Movies/gizmo2.mp4'
+        }
+      ]);
+    });
+
+    test('Check the mp4 video isn\'t loaded', function() {
+
+      music.launch();
+      music.waitForFirstTile();
+      music.waitFinishedScanning();
+
+      music.switchToSongsView();
+
+      music.waitForSongsInList(function(songs) {
+        return songs.length >= 2;
+      });
```

Comment:
I don't like this because you're waiting for there two be two songs, then testing that there are only two songs.  If a third song is added to the list later, then the test could pass when it should have failed.

Given the music app's current architecture, I don't see an easy way to know when the songs view is fully populated. Maybe you could modify switchToSongsView() so that it patches App.showCorrectOverlay() and use that to detect when the songs view is full loaded.

If you don't do that, I'd like you to at least add a comment here about the issue and wait for an extra second to make sure that no more songs are added.

Actually, here's a better way: when list_view.js enumerates the db, it sets this.handle to the return value of the enumerate() call.  this.handle is an object with a state property.  If you wait for `this.handle.state === 'complete'` then you'll know that the enumeration is complete.